src/Frontend/src/stores/ServiceControlStore.ts

```diff
@@ -0,0 +1,191 @@
+import { acceptHMRUpdate, defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+export const useServiceControlStore = defineStore("ServiceControlStore", () => {
+  const serviceControlUrl = ref<string | null>();
+  const monitoringUrl = ref<string | null>();
+
+  const isMonitoringDisabled = computed(() => monitoringUrl.value == null || monitoringUrl.value === "" || monitoringUrl.value === "!");
+  const isMonitoringEnabled = computed(() => !isMonitoringDisabled.value);
+
+  function getServiceControlUrl() {
+    if (!serviceControlUrl.value) refresh();
```

Comment:
We should throw here to prevent weird behaviour later on.
```suggestion
    if (!serviceControlUrl.value) {
       refresh();
    }
    if (!serviceControlUrl.value) {
      throw new Error("Service Control URL is not configured");
    }
```